mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -186,10 +198,14 @@ pub struct Nexus {
     /// the offset in num blocks where the data partition starts
     pub data_ent_offset: u64,
     /// nbd device which the nexus is exposed through
-    pub(crate) nbd_disk: Option<Disk>,
+    pub(crate) nbd_disk: Option<NbdDisk>,
     /// the handle to be used when sharing the nexus, this allows for the bdev
     /// to be shared with vbdevs on top
     pub(crate) share_handle: Option<String>,
+    /// frontend share protocol used when the nexus is published
+    pub share_protocol: Option<ShareProtocolNexus>,
```

Comment:
If the nexus is not shared, I assume this share_protocol would have the variant, so we really don't need to put this in an option, it should always have a value, right?